Clockwise/ConfigurationExtensions.cs

```diff
@@ -50,6 +51,66 @@ public static class ConfigurationExtensions
             return configuration;
         }
 
+        public static Configuration UseCircuitbreaker<TChannel, TCircuitBreaker>(this Configuration configuration) where TCircuitBreaker : ICircuitBreaker
+        {
+            configuration.Container.AfterCreating<ICommandReceiver<TChannel>>(receiver =>
+            {
+                TCircuitBreaker cb;
+                try
+                {
+                    cb = configuration.Container.Resolve<TCircuitBreaker>();
+                    
+                }
+                catch (Exception e)
+                {
+                    throw new ConfigurationException($"Failure during creation of circuit breaker {typeof(TCircuitBreaker).Name}",e);
+                }
+
+                async Task<ICommandDeliveryResult> Receive(HandleCommand<TChannel> handlerDelegate, TimeSpan? timeout, Func<HandleCommand<TChannel>, TimeSpan?, Task<ICommandDeliveryResult>> next)
```

Comment:
Here's another `next` that would like to be renamed.

Reply:
Done